pandas/_libs/index.pyx

```diff
@@ -275,7 +275,11 @@ cdef class IndexEngine:
 
         self._ensure_mapping_populated()
         values = np.array(self._get_index_values(), copy=False)
+        values_mask = isnaobj(values)
+        targets_mask = isnaobj(targets)
         stargets = set(targets)
+        if targets_mask.any():
+            stargets.add(-1)
```

Comment:
do we know that -1 isnt already in stargets?  or does it not matter?

Reply:
I can revise to `targets_mask.any() and -1 not in stargets`. however, I don't think it matters because the end result will be the same

Reply:
Well if we know then we can consolidate this whole if else no?

Reply:
@arw2019 by consolidate, do you mean checking for -1 in the set before checking for nans? like what I wrote above?

Reply:
As long as -1 doesn't need extra handling we can just delete this if statement 

I see you already consolidated

Reply:
I think we'll still need this if statement. consider the following scenario:

```
>>> import numpy as np
>>> targets = [np.nan]
>>> stargets = set(targets)
>>> stargets
{nan}
```

we would need to add -1 in stargets to ensure this line works:

```
>>> val = -1
>>> if val in stargets:
...     print("in stargets")
... 
```

Reply:
I kept the if statement and added a check for -1 in my most recent commit - does this work?

Reply:
yeah i think so